crates/federate/src/worker.rs

```diff
@@ -265,7 +265,7 @@ impl InstanceWorker {
         SendActivityResult::Success(s) => {
           self.in_flight -= 1;
           if !s.was_skipped {
-            self.state.fail_count = 0;
+            self.state.fail_count = max(0, self.state.fail_count - 1);
```

Comment:
Just in case `fail_count` becomes unsigned in the future

```suggestion
            self.state.fail_count = max(1, self.state.fail_count) - 1;
```

Reply:
I just realized that a test would catch the underflow, so this is acceptable either way, as long as the test coverage is kept.